From 87e9ef7e32bc1b4147b1c8a5b4bc02fe6bf08dc9 Mon Sep 17 00:00:00 2001
From: Amarnath Valluri <amarnath.valluri@intel.com>
Date: Wed, 16 Mar 2016 01:29:59 +0200
Subject: [PATCH] build.go: Pass CGO_CFLAGS for building .c files

Upstream-Status: Not-Submitted

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>
---
 src/cmd/go/build.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cmd/go/build.go b/src/cmd/go/build.go
index f2a2a60..3cbd2df 100644
--- a/src/cmd/go/build.go
+++ b/src/cmd/go/build.go
@@ -2819,8 +2819,9 @@ func (tools gccgoToolchain) cc(b *builder, p *Package, objdir, ofile, cfile stri
 		defs = append(defs, "-fsplit-stack")
 	}
 	defs = tools.maybePIC(defs)
+  _, cgoCFLAGS, _, _ := b.cflags(p, true)
 	return b.run(p.Dir, p.ImportPath, nil, envList("CC", defaultCC), "-Wall", "-g",
-		"-I", objdir, "-I", inc, "-o", ofile, defs, "-c", cfile)
+		"-I", objdir, "-I", inc, "-o", ofile, defs, cgoCFLAGS, "-c", cfile)
 }
 
 // maybePIC adds -fPIC to the list of arguments if needed.
-- 
1.9.4

